pkg/build/controller/strategy/util.go

```diff
@@ -90,3 +65,49 @@ func setupBuildEnv(build *buildapi.Build, pod *kapi.Pod) error {
 	}
 	return nil
 }
+
+// setupDockerSecrets mounts Docker Registry secrets into Pod running the build,
+// allowing Docker to authenticate against private registries or Docker Hub.
+func setupDockerSecrets(pod *kapi.Pod, secretRef *kapi.ObjectReference) {
+	if secretRef == nil {
+		return
+	}
+
+	// TODO: Should namespace be required here?
+	namespace := secretRef.Namespace
+	if len(namespace) == 0 {
+		namespace = pod.Namespace
+	}
+
+	if len(namespace) == 0 {
+		glog.Infof("Unable to get namespace from RegistrySecret, using default")
+		namespace = kapi.NamespaceDefault
```

Comment:
@liggitt I fixed this in last commit so this `if` is just a fallback (which is why it emits the message to log). The `pod.Namespace` should be set to `build.Namespace` now.